src/algebra/graded_monoid.lean

```diff
@@ -201,6 +213,34 @@ instance gcomm_monoid.to_comm_monoid [add_comm_monoid ι] [gcomm_monoid A] :
   comm_monoid (graded_monoid A) :=
 { mul_comm := gcomm_monoid.mul_comm, ..gmonoid.to_monoid A }
 
+/-- A graded version of `mul_action`. -/
+class gmul_action [add_monoid ι] [gmonoid A] extends ghas_smul A M :=
+(one_smul (b : graded_monoid M) : (1 : graded_monoid A) • b = b)
+(mul_smul (a a' : graded_monoid A) (b : graded_monoid M) : (a * a') • b = a • a' • b)
```

Comment:
I wonder why you don't restrict `a, a', b` to some `A i` / `M j` like in `gdistrib_mul_action` and `g,odule` below. Is that not enough?

Reply:
For type checking reasons. For example for `b : M j` `smul 1 b` is of type `M (0 + j)`, so can't be compared with `b` anymore.

Reply:
The relevant comparison here is to `gmonoid`.